caffe2/serialize/inline_container.cc

```diff
@@ -42,27 +46,31 @@ static std::string basename(const std::string& name) {
 }
 
 size_t PyTorchStreamReader::read(uint64_t pos, char* buf, size_t n) {
-  in_->seekg(pos);
-  if(!*in_)
-    return 0;
-  in_->read(static_cast<char*>(buf), n);
-  if(!*in_)
-    return 0;
-  return n;
+  return in_->read(pos, buf, n, "reading file");
+}
+
+PyTorchStreamReader::PyTorchStreamReader(std::string file_name, std::istream* in) {
```

Comment:
btw, when `in` is passed, file_name is ignored. Should it be two separate constructors? (it was also in the original code)

Reply:
Yeah, it can be deleted. An old version of the code needed to know the filename to know the top-level folder in archive where everything is, but now it just reads that from the first file in the archive.